src/librustc/traits/select.rs

```diff
@@ -1884,6 +1872,17 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
                     }
                 }
                 ty::TyInfer(ty::TyVar(_)) => {
+                    // Object-safety candidates are only applicable to object-safe
```

Comment:
Does this check do anything these days? I think `$0: Trait` is always considered ambiguous.

Reply:
Good point @arielb1 -- we may be able to just remove this code altogether.